### PR TITLE
Adding Cloud Backup endpoints.

### PR DESCRIPTION
**Pull request by rsmartins78**
Added:
- Ability to list all snapshots.
- Get one snapshot by ID.
- Request a restore of the snapshot to a target.

Sorry for any mistake, I'm not JS specialist.

### CODE REVIEW

File: src/cloudBackup.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const {getQueryStringFromOptions} = require("mongodb-atlas-api-client/src/helper");
```
Comment (montumodi) [inline src/cloudBackup.js:1]: This should be `const {getQueryStringFromOptions} = require("./helper");
`

File: src/cloudBackup.d.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    oplogInc?: number;
+    oplogTs?: number;
+    pointInTimeUTCSeconds?: number;
+    snapshotId?: string;
+    targetClusterName?: string;
+    targetGroupId?: string
+}
+export type CreateRestoreSnapshotJobResponse = GetClusterResponse;
+
+export interface Cluster {
```
Comment (montumodi) [inline src/cloudBackup.d.ts:65]: Should this be Cluster of Cloudbackup?

**Comment by montumodi** [comment]
Thanks for raising the PR. Could you please update following
- Readme : with the new method names and links
- Tests - At the moment code coverage is below 100% and hence it is failing. You can follow tests written for other classes as a ref.

File: src/cloudBackup.js (hunk 1 above)
Comment (rsmartins78) [reply src/cloudBackup.js]: fixed

File: src/cloudBackup.d.ts (hunk 2 above)
Comment (rsmartins78) [reply src/cloudBackup.d.ts]: fixed

**Comment by rsmartins78** [comment]
> Thanks for raising the PR. Could you please update following
> 
> * Readme : with the new method names and links
> * Tests - At the moment code coverage is below 100% and hence it is failing. You can follow tests written for other classes as a ref.

added tests

**Comment by montumodi** [comment]
Pushed a new [minor ](https://www.npmjs.com/package/mongodb-atlas-api-client/v/3.19.0) version.

**Pull request merged**